firmware/src/widgets/stockWidget.cpp

```diff
@@ -108,29 +112,66 @@ void StockWidget::displayStock(int8_t displayIndex, StockDataModel &stock, uint3
     TFT_eSPI &display = m_manager.getDisplay();
 
     display.fillScreen(TFT_BLACK);
-    display.setTextColor(TFT_WHITE);
-    display.setTextSize(4);  // Set text size
+
 
     // Calculate center positions
     int screenWidth = display.width();
     int centre = 120;
-
-    // Draw stock data
-    display.fillRect(0, 0, screenWidth, 50, 0x0256);  // rgb565 colors
-    display.drawString(stock.getSymbol(), centre, 27, 1);
-    display.drawString("$" + stock.getCurrentPrice(2), centre, 51 + display.fontHeight(1), 1);
-
+    int arrowOffsetX = 0;
+    int arrowOffsetY = -109;
+
+    // ensuring proper currency display
+    String currencySymbol = "$";
+    String currencySymbolEUR = "EUR";
+    String currencySymbolGBP = "GBP";
+    // the below are just placeholders untill we have a unicode font that supports the currency symbols, which is why we're not just using the direct value from getCurrency.
+    // additionally, in some cases(crypto) the ticker wont return a currency value, so it must be extracted from the symbol deffinition 
+    if (stock.getCurrency() == "EUR"){
+       currencySymbol = currencySymbolEUR;
+    } else if (stock.getCurrency() == "GBP"){
+       currencySymbol = currencySymbolGBP;
+    } else if (stock.getSymbol().indexOf("/EUR") != -1){
+       currencySymbol = currencySymbolEUR;
+    } else if (stock.getSymbol().indexOf("/GBP") != -1){
+       currencySymbol = currencySymbolGBP;
+    }
+// Outputs
+    display.fillRect(0,70, screenWidth, 49, TFT_WHITE); 
+    display.fillRect(0,111, screenWidth, 15, TFT_LIGHTGREY);  
+    display.setTextSize(1);  
+    display.setTextColor(TFT_WHITE);
+    display.drawString("52 Week:", centre, 188, 1);
+    display.setTextDatum(ML_DATUM);
+    display.drawString("H: " + currencySymbol + stock.getHighPrice(), 89, 201, 1);
+    display.drawString("L: " + currencySymbol + stock.getLowPrice(), 89, 211, 1);
+    display.setTextDatum(CC_DATUM);
+    display.setTextColor(TFT_BLACK);
+    display.drawString(stock.getCompany(), centre, 118, 1);
+    display.setTextSize(4); 
     if (stock.getPercentChange() < 0.0) {
```

Comment:
Is this `getPercentChange` more relevant than a trend?
(I am unsure if this api has a trend)

Reply:
Its standard in the stock market for performance to be tracked day/day, trend would be possible, but thats just not how most folks that follow stocks closely expect the data to be represented

Reply:
agreed, a trend is a separate thing. Percent change is to help you see that a $1.00 stock going up $0.50 is a much bigger deal than a $1000 stock going up $0.50. So it is easy to show. But to show a trend you really need a graph or something that shows more data points and a time frame over which you are looking at the trend. If someone was really into it we could do a stock view that rotated between stocks with one stock data filling all 5 orbs.